lib/cretonne/meta/cdsl/registers.py

```diff
@@ -299,19 +330,13 @@ def mask(self):
 
         Return as a list of 32-bit integers.
         """
-        mask = [0] * MASK_LEN
-
-        start = self.bank.first_unit + self.start
-        for a in range(self.count):
-            u = start + a * self.width
-            b = u % 32
-            # We need fancier masking code if a register can straddle mask
-            # words. This will only happen with widths that are not powers of
-            # two.
-            assert b + self.width <= 32, 'Register straddles words'
-            mask[u // 32] |= 1 << b
-
-        return mask
+        out_mask = []
+        mask32 = (1 << 32) - 1
+        bitmask = self.bitmask << self.bank.first_unit
+        for i in range(3):
```

Comment:
This `3` here is kind of a magic number. Could we avoid it by doing something like this?
```python
    while bitmask != 0:
        ... // do stuff with bitmask & mask32
        bitmask >>= 32
```

Reply:
The problem with that is that the Rust array this code is used to generate is expected to always have a length of 3. (https://github.com/stoklund/cretonne/blob/master/lib/cretonne/src/isa/registers.rs#L15-L21) However, there is a constant for this that I forgot to use: `MASK_LEN`. Fixing that.